doc/called_by.md

```diff
@@ -0,0 +1,35 @@
+# Reference kinds
+
+## Short introduction
+
+This implements the functionality 'is called by', allowing to query
+all the calls to a function, organized by the entities where these
+calls happen.
+
+## Change description
+
+We introduce a new type to represent the results:
+
+```typescript
+
+interface ALSSubprogramAndReferences {
+
+   loc: Location; /* The location of the entity containing the references */
+   name: string; /* The name of the entity containing the references */
+
+   refs: Location[];  /* The references contained in this entity */
+}
+```
+
+And a new request:
+
+  method: `textDocument/ALS_called_by`
```

Comment:
Shall we follow "camelCase" naming on method names here? See, for example `textDocument/onTypeFormatting`.

I propose `textDocument/alsCalledBy`. What do you think?

Reply:
:+1: for me